distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestVideo.java

```diff
@@ -0,0 +1,226 @@
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+import com.jayway.restassured.response.ValidatableResponse;
+
+import ddf.common.test.BeforeExam;
+import ddf.test.itests.AbstractIntegrationTest;
+
+/**
+ * Tests Alliance video capabilities.
+ */
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class TestVideo extends AbstractIntegrationTest {
+    private static final String[] REQUIRED_APPS = {"catalog-app", "solr-app", "video-app"};
+
+    private static final String METACARD_COUNT_XPATH = "count(/metacards/metacard)";
+
+    private static final String METACARD_ID_XMLPATH = "metacards.metacard.@gml:id";
+
+    private static final String DAYFLIGHT = "dayflight.mpg";
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private DynamicPort udpPort;
+
+    @BeforeExam
+    public void beforeExam() throws Exception {
+        basePort = getBasePort();
+        udpPort = new DynamicPort(6);
+
+        getServiceManager().waitForRequiredApps(REQUIRED_APPS);
+        getServiceManager().waitForAllBundles();
+        getCatalogBundle().waitForCatalogProvider();
+    }
+
+    @Override
+    protected Option[] configureDistribution() {
+        return options(karafDistributionConfiguration(maven().groupId(
+                "org.codice.alliance.distribution")
+                .artifactId("alliance")
+                .type("zip")
+                .versionAsInProject()
+                .getURL(), "alliance", KARAF_VERSION).unpackDirectory(new File("target/exam"))
+                .useDeployFolder(false));
+    }
+
+    @Override
+    protected Option[] configureCustom() {
+        return options(wrappedBundle(mavenBundle("ddf.test.itests", "test-itests-ddf").classifier(
+                "tests")
+                        .versionAsInProject()).bundleSymbolicName("test-itests-ddf")
+                        .exports("ddf.test.itests.*"),
+                wrappedBundle(mavenBundle("org.codice.alliance.distribution",
+                        "sample-mpegts-streamgenerator").versionAsInProject()),
+                wrappedBundle(mavenBundle("io.netty", "netty-all", "4.0.36.Final")),
+                wrappedBundle(mavenBundle("org.apache.commons", "commons-exec", "1.2")),
+                wrappedBundle(mavenBundle("org.apache.commons", "commons-lang3", "3.4")));
+    }
+
+    @Test
+    public void testStreamingVideo() throws Exception {
+        final String videoFilePath = FilenameUtils.concat(temporaryFolder.getRoot()
+                .getCanonicalPath(), DAYFLIGHT);
+        final File videoFile = new File(videoFilePath);
+
+        copyResourceToFile(DAYFLIGHT, videoFile);
+
+        startUdpStreamMonitor();
+
+        final int udpPortNum = Integer.parseInt(udpPort.getPort());
+
+        expect("The UDP stream monitor to start on port " + udpPort.getPort()).within(5,
```

Comment:
It's necessary to wait for the UDP stream monitor to bind to the port before we start broadcasting the video (this ensures that the monitor will not miss any of the video bytes).